spatial/r3/affine.go

```diff
@@ -0,0 +1,255 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3
+
+import (
+	"fmt"
+	"math"
+)
+
+// Affine represents a 3D affine spatial transformation.
+// The zero value of Affine is the identity transform.
+type Affine struct {
+	// in order to make the zero value of Transform represent the identity
+	// transform we store it with the identity matrix subtracted.
+	// These diagonal elements are subtracted such that
+	//  d00 = x00-1, d11 = x11-1, d22 = x22-1, d33 = x33-1
+	// where x00, x11, x22, x33 are the matrix diagonal elements.
+	// We can then check for identity in if blocks like so:
+	//  if T == (Transform{})
+	d00, x01, x02, x03 float64
+	x10, d11, x12, x13 float64
+	x20, x21, d22, x23 float64
+	x30, x31, x32, d33 float64
+}
+
+// Transform applies the transform to the argument vector and returns the result.
+func (a Affine) Transform(v Vec) Vec {
+	if a.isIdentity() {
+		return v
+	}
+	// See https://github.com/mrdoob/three.js/blob/019fa1ad671a1ffcf9be5828efd518fb06575c2b/src/math/Vector3.js#L263.
+	w := 1 / (a.x30*v.X + a.x31*v.Y + a.x32*v.Z + a.d33 + 1)
```

Comment:
What kind of transformations does Affine represent? The 4x4 nature and the code here suggest that perspective projection is included. Is it intended? I'd say maybe it's out of scope?